torch/_dynamo/backends/distributed.py

```diff
@@ -445,10 +454,10 @@ def run_node(self, n: Node) -> Any:
                     # Finally, we have to produce inputs for use compiling the next submodule,
                     # and these need to be FakeTensors, so we execute the module under fake_mode
                     with fake_mode:
-                        return curr_submod(*new_args, **kwargs)
+                        return curr_submod(*fake_args, **kwargs)
                 else:
                     # placeholder or output nodes don't need to get compiled, just executed
-                    return getattr(self, n.op)(n.target, new_args, kwargs)
+                    return getattr(self, n.op)(n.target, fake_args, kwargs)
 
         submod_compiler = SubmodCompiler(split_gm, self.backend_compile_fn)
         submod_compiler.run(*example_inputs)
```

Comment:
Logging the final graph below doesn't seem so meaningful anymore. Maybe only log it when the entire graph module is run?